### DISCUSSION TITLE
Does size matter... in the metadata?

### DISCUSSION THREAD

**Discussion by Adamska** [capsule-corp-ternoa/ternoa-hub#26] [General]
Hi, in the metadata standard of an NFT, there is a `size` field in the `media` object. Is it a mandatory field and what is it used for ?

I am asking that because there are 2 cases when generating the metadata with a script :
- 1st case, you upload the files in your script while generating the metadata, so you can get the file size easily, no problem here
- 2nd case, you already uploaded your files on a IPFS storage provider, let's say Pinata or NFT storage, in this case it is a pain in the ass to have to download every file just to get their filesize (unless you know of a mean to do so, I didn't find any in my searches). If the field appears to be optional, that would be really easier to many projects that build and generate metadata after having uploaded their assets on an IPFS storage provider

**Comment by peshwar9** [comment]
@Adamska : The size property (and any other offchain metadata) is suggested as a recommendation. The blockchain, as you are aware, cannot enforce it. But the reason size property was included in offchain metadata was to future-proof it, when there could be multiple NFT data backup providers, and having an idea of size of data can help in negos with data backup service providers.
@mohsinriaz17 : Do you have anything to add about the size property?
